lib/ae_mdw_web/controllers/name_controller.ex

```diff
@@ -229,28 +234,40 @@ defmodule AeMdwWeb.NameController do
   end
 
   @spec search(Conn.t(), map()) :: Conn.t()
-  def search(conn, %{"prefix" => prefix}) do
-    handle_input(conn, fn ->
-      params = Map.put(query_groups(conn.query_string), "prefix", [prefix])
-      json(conn, Enum.to_list(do_prefix_stream(validate_search_params!(params), expand?(params))))
-    end)
-  end
-
-  ##########
-
-  # scope is used here only for identification of the continuation
-  @spec db_stream(atom(), map(), term()) :: Enumerable.t()
-  def db_stream(:auctions, params, _scope),
-    do: do_auctions_stream(validate_params!(params), expand?(params))
+  def search(%Conn{request_path: path, assigns: assigns, query_string: query_string} = conn, %{
+        "prefix" => prefix
+      }) do
+    lifecycles =
+      query_string
+      |> URI.query_decoder()
+      |> Enum.filter(&match?({"only", _lifecycle}, &1))
+      |> Enum.map(fn {"only", lifecycle} -> lifecycle end)
+      |> Enum.filter(&Map.get(@lifecycles_map, &1))
```

Comment:
maybe join this filter with previous one

Reply:
Done